backend/common/utils/utils.py

```diff
@@ -78,9 +78,9 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     def default(self, obj):
-        if isinstance(obj, np.float32):
+        if np.issubdtype(obj, np.floating) and not isinstance(obj, np.float16):
```

Comment:
I don't understand how this is changing the existing behavior, since you are still gating it to float32 (albeit adding in support for 64 and 128). How will float16s get encoded?

Reply:
oh my, this code has so many issues, I can't believe I actually committed it.   Will fix.